x/gamm/keeper/swap.go

```diff
@@ -154,22 +154,24 @@ func (k Keeper) updatePoolForSwap(
 		return err
```

Comment:
I don't think calling a setter method(`setPool`) inside a query method is a safe way of doing this IMO.
We should either completely separate out the querying methods from the existing method or create a new separate method for this estimate call

Reply:
Yeah, this is concerning, great catch.

So turns out some queries in the SDK already do this, so for now this is grudgingly ok, we should cache the context in our query layer to prevent this for now.

Reply:
I suggest we don't block this pr on this